src/pybind/mgr/dashboard/controllers/rbd.py

```diff
@@ -231,6 +231,9 @@ class RbdSnapshot(RESTController):
     @RbdTask('snap/create',
              ['{image_spec}', '{snapshot_name}', '{mirrorImageSnapshot}'], 2.0)
     def create(self, image_spec, snapshot_name, mirrorImageSnapshot):
+        if snapshot_name == "." or "/" in snapshot_name:
+         raise cherrypy.HTTPError(400, 'Snapshot name "." or "/" is not allowed.')
```

Comment:
I saw the RBD team is using a pattern to disallow `/` and `@` in the snap name. Ideally our API as a consumer of librbd interface, should throw the error directly from the librbd for the validation failure. Probably the librbd interface itself is not validating it? Maybe @idryomov can confirm it..


But in any case our UI can try to use the pattern that [we see in the rbd side](https://github.com/ceph/ceph/blob/main/src/tools/rbd/Utils.cc#L122).